### DISCUSSION TITLE
Solve diffusion equation and Laplace equation together

### DISCUSSION THREAD

**Discussion by th509** [zwicker-group/py-pde#198] [Q&A] [answered]
Thanks for your contribution!

I want to solve diffusion equation with electro-migration term.

![Electro-migration](https://latex.codecogs.com/gif.latex?%5Cfrac%7B%5Cpartial%20c%7D%7B%5Cpartial%20t%7D%20%3D%20D%20%5Cnabla%5E2%28c&plus;c%5Cphi%29)

Current code with custom pde class is written without electromigration term

```
def evolution_rate(self, state, t=0):
   return self.diffusivity * state.laplace(bc=["periodic", [{"value": 0}, {"value": 1}]])
```

The electric potential (\phi) should be solved simultaneously following equation,

![Electric potential](https://latex.codecogs.com/gif.latex?%5Cnabla%20%5B%28%5Cepsilon_1%20%5Cfrac%7B%28c-c_1%29%5E2%7D%7B%28c_2-c_1%29%5E2%7D%20&plus;%20%5Cepsilon_2%20%5Cfrac%7B%28c-c_2%29%5E2%7D%7B%28c_1-c_2%29%5E2%7D%29%20%5Cnabla%20%5Cphi%5D%20%3D%200)

where c1 = 0.1 and c2 = 0.9, epsilon2 = epsilon1 * 1000

How can I solve the time-dependent diffusion equation and steady state Laplace equation together?

**Comment by david-zwicker** [comment, marked as answer]
I'm afraid that this is not easily possible in the current version of the package since it focuses more on solving dynamical equations (whereas in your case \phi is not a dynamical field anymore).
